projects/core/src/anonymous-consents/services/anonymous-consents-state-persistence.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable, Subscription } from 'rxjs';
+import { StatePersistenceService } from '../../state/index';
+import { AnonymousConsentsService } from '../facade/index';
+import { LoadAnonymousConsentTemplatesSuccess } from '../store/actions/anonymous-consents-group';
+import {
+  AnonymousConsentsState,
+  StateWithAnonymousConsents,
+} from '../store/index';
+import { getAnonymousConsentState } from '../store/selectors/feature.selector';
+
+/**
+ * Anonymous consents state synced to browser storage.
+ */
+export type SyncedAnonymousConsentsState = Partial<AnonymousConsentsState>;
+
+/**
+ * Responsible for saving the anonymous consents data in browser storage.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AnonymousConsentsStatePersistenceService implements OnDestroy {
+  protected subscription = new Subscription();
+
+  constructor(
+    protected statePersistenceService: StatePersistenceService,
+    protected store: Store<StateWithAnonymousConsents>,
+    protected anonymousConsentsService: AnonymousConsentsService
+  ) {}
+
+  /**
+   * Identifier used for storage key.
+   */
+  protected key = 'anonymous-consents';
+
+  /**
+   * Initializes the synchronization between state and browser storage.
+   */
+  public initSync() {
+    this.subscription.add(
+      this.statePersistenceService.syncWithStorage({
+        key: this.key,
+        state$: this.getAuthState(),
+        onRead: (state) => this.onRead(state),
+      })
+    );
+  }
+
+  /**
+   * Gets and transforms state from different sources into the form that should
+   * be saved in storage.
+   */
+  protected getAuthState(): Observable<SyncedAnonymousConsentsState> {
+    return this.store.select(getAnonymousConsentState);
+  }
+
+  /**
+   * Function called on each browser storage read.
+   * Used to update state from browser -> state.
+   */
+  protected onRead(state: SyncedAnonymousConsentsState) {
+    if (state) {
+      if (state.templates) {
```

Comment:
One more thing just to be safe. Lets check here as well if we got success state. Otherwise there is no point to restore it as it might even break something.
```suggestion
      if (state.templates && state.templates.success) {
```